addon/transforms/string.ts

```diff
@@ -24,10 +24,10 @@ import Transform from './transform';
   @namespace DS
  */
 export default Transform.extend({
-  deserialize(serialized) {
+  deserialize(serialized: any) {
```

Comment:
Are `null`, `undefined` and non-serializable objects (i.e., circular references) valid things to pass here?

Reply:
unsure, I started this file as the first TS file just to see if I could get a build/tests going.

One of the things I'm excited about with the TS conversion, is that anybody will be able to answer that question :)